src/main/java/util/TaskList.java

```diff
@@ -0,0 +1,167 @@
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+
+
+/**
+ * TaskList class to help manage Tasks.
+ * @author Merrick
+ */
+public class TaskList {
+    private static final int TASK_TYPE = 0;
+    private static final int TASK_NAME = 1;
+    private static final int IS_DONE = 2;
+    protected ArrayList<Task> taskList = new ArrayList<>();
+    protected BufferedReader br;
+
+    /**
+     * Empty constructor for TaskList.
+     */
+    public TaskList() {}
+
+    /**
+     * Constructor of TaskList with tasks loaded from storage.
+     * @param br Saved-data loaded stored in BufferedReader object.
+     * @throws DukeException If data is unable to be read from BufferedReader object br.
+     */
+    public TaskList(BufferedReader br) throws DukeException {
+        this.br = br;
+        this.load();
+    }
+
+    /**
+     * Adds a new task to be tracked by TaskList.
+     * <br>
+     * Displays the number of tasks after addition of task.
+     * @param task Task object to be tracked.
+     */
+    public void addTask(Task task) {
+        taskList.add(task);
+        System.out.printf("     Got it. I've added this task:\n"
+                + "       %s\n" + this.numTasks() + "%n", task);
+    }
+
+    /**
+     * Displays the tasks being tracked by TaskList.
+     */
+    public void listTasks() {
+        System.out.println("    Here are the tasks in your list:");
+        int counter = 1;
+        for (Task t : taskList) {
+            System.out.println(counter + ". " + t.toString());
+            counter++;
+        }
+    }
+
+    /**
+     * Displays the tasks in the TaskList that contains the keyword specified.
+     * @param command Keyword to check with the tasks in the TaskList.
+     */
+    public void findTasks(String command) {
+        System.out.println("  Here are the matching tasks in your list:");
+        String[] keyword = command.split(" ");
+        int counter = 1;
+        for (Task t: taskList) {
+            // Iterate through the ArrayList to find if a Task matches
+            if (t.containsKeyword(keyword[1])) {
+                System.out.println(counter + ". " + t.toString());
+                counter++;
+            }
+        }
+
+    }
+
+    /**
+     * Loads the tasks stored in the BufferedReader object into the task list.
+     * @throws DukeException If IOException from the BufferedReader object is encountered.
+     */
+    @SuppressWarnings("checkstyle:RightCurly")
+    public void load() throws DukeException {
+        try {
+            while (true) {
+                String taskLine = br.readLine();
+                if (taskLine == null) {
+                    break;
+                }
+                String[] taskArr = taskLine.split("\\|");
+                String taskType = taskArr[TASK_TYPE];
+                switch (taskType) {
+                case "T":
+                    boolean completion = Boolean.parseBoolean(taskArr[IS_DONE]);
```

Comment:
Would be better to rename this boolean variable to `isCompleted` or `hasCompleted`

Reply:
Thanks for the spot!